src/env.cc

```diff
@@ -181,6 +193,26 @@ void IsolateData::CreateProperties() {
         sizeof(#Provider) - 1).ToLocalChecked());
   NODE_ASYNC_PROVIDER_TYPES(V)
 #undef V
+
+  // Create all the hook type strings that will be passed to JS. Place them in
+  // an array so the array index matches the type id offset. This way the
+  // strings can be retrieved quickly.
+#define PROMISE_HOOK_TYPES(V)                                                 \
```

Comment:
This enum seems to have different order (init, promiseResolve, before, after). See:
https://github.com/nodejs/node/blob/master/deps/v8/include/v8.h#L7264

I tried to change it locally and it fixed most of tests in `test/async-hooks/`.

Reply:
Gah...of course the issue was something silly like that. Good catch. 😬